lib/ansible/modules/find.py

```diff
@@ -530,7 +553,8 @@ def handle_walk_errors(e):
             if not os.path.isdir(npath):
                 raise Exception("'%s' is not a directory" % to_native(npath))
 
-            for root, dirs, files in os.walk(npath, onerror=handle_walk_errors, followlinks=params['follow']):
+            # Setting `topdown=True` to explicitly guarantee matches are made from the shallowest directory first
+            for root, dirs, files in os.walk(npath, onerror=handle_walk_errors, followlinks=params['follow'], topdown=True):
```

Comment:
this changes the normal search by adding topdown, this is not really backwards compatible

Reply:
i would even add this as an option before i limit matching

Reply:
`topdown` is `True` by default (https://docs.python.org/3/library/os.html#os.walk) so it should not change the current behaviour. The intention was to make it explicit that the module is guaranteed to be topdown - perhaps the docs is a better place for this though?

I'm not sure how many people will have a use-case where they'd want to switch between topdown/bottomup. However, I'm happy to add it if you think it could be useful.